meso_tools/NAS_tools.py

```diff
@@ -0,0 +1,96 @@
+    res = {keys[i]: values[i] for i in range(len(keys))}
+    return(res)
+
+def nas_query(credentials: dict, nas_name: str, release_list: str):
+    """Uses requests to login to the NAS storage, query the list of folders being backed up, and return a list of
+    folders that have been released by the AllenSDK
+
+    Args:
+        credentials (dict): output from credentials_read function containing Synology NAS credentials
+        nas_name (str): name of the storage servers (currently either 'ophys_nas' or 'ophys_storage')
+        release_list (str): Relative string filepath to a .csv file containing information about released ophys sessions. 
+        This data is the result of an AllenSDK query and not currently included in this code
+    """    
+
+    user = credentials['SYNOLOGY_USERNAME']
+    password = credentials['SYNOLOGY_PASSWORD']
+    ip = credentials['SYNOLOGY_IP']
+
+    # login
+    response = requests.get("http://"+str(ip)+"/webapi/auth.cgi?api=SYNO.API.Auth&version=2&method=login&account="+str(user)+ 
+        "&passwd="+str(password)+"&session=FileStation&format=sid")
+    sid = (response.json()['data']['sid'])
+    if nas_name == 'ophys_nas':
+        #folder 1
+        response = requests.get("http://"+str(ip)+"/webapi/entry.cgi?api=SYNO.FileStation.List&version=2&method=list&additional=%5B%22real_path%22%2C%22size%2Cperm%22%5D&folder_path=%22%2Fmeso2_backup/Data_Backup%22&_sid="+sid)        
+        released_backups = release_check(release_list, response)
+        
+        #folder 2
+        response = requests.get("http://"+str(ip)+"/webapi/entry.cgi?api=SYNO.FileStation.List&version=2&method=list&additional=%5B%22real_path%22%2C%22size%2Cperm%22%5D&folder_path=%22%2Fmeso1_backup/Data_Backup%22&_sid="+sid)
+        released_backups2 = release_check(release_list, response)
+    elif nas_name == 'ophys_backup':
+
+        #folder 1
+        response = requests.get("http://"+str(ip)+"/webapi/entry.cgi?api=SYNO.FileStation.List&version=2&method=list&additional=%5B%22real_path%22%2C%22size%2Cperm%22%5D&folder_path=%22%2Fmeso_backup_full/data_backup%22&_sid="+sid)
+        released_backups = release_check(release_list, response)
+        #folder 2
+        response = requests.get("http://"+str(ip)+"/webapi/entry.cgi?api=SYNO.FileStation.List&version=2&method=list&additional=%5B%22real_path%22%2C%22size%2Cperm%22%5D&folder_path=%22%2Fmeso_backup/data_backup%22&_sid="+sid)
+        released_backups2 = release_check(release_list, response)          
+    else:
+        print('NAS name not valid')
+        released_backups = []
+        released_backups2 = []
+    # logout
+
+    requests.get("http://"+str(ip)+"/webapi/auth.cgi?api=SYNO.API.Auth&version=6&method=logout&session=FileStation")
+    return(released_backups, released_backups2)
+            
+def release_check(release_list: str, query_response: requests.models.Response):
+    """Compares the output of a requests query against a list of released sessions, 
```

Comment:
Let's use Numpy format for docstrings - this is set is autodocstring plugin settings